src/create-broadcast.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * Creates a broadcast that can be listened to, i.e. simple event emitter
+ *
+ * @see https://github.com/ReactTraining/react-broadcast
+ */
+
+const createBroadcast = initialValue => {
+  let listeners = [];
+  let currentValue = initialValue;
+
+  return {
+    publish(value) {
+      currentValue = value;
+      listeners.forEach(listener => listener(currentValue));
+    },
+    subscribe(listener) {
+      listeners.push(listener);
+
+      // Publish to this subscriber once immediately.
+      listener(currentValue);
+
+      return () => {
+        listeners = listeners.filter(item => item !== listener);
+      };
+    },
+  };
+};
+
+module.exports = createBroadcast;
```

Comment:
@iamstarkov `componentDidMount` is used only to create subscriptions. When `componentWillMount` is executed on the server, it applies the theme (in this phase you do not care about subscriptions). Then on the client subscriptions are created and theme is now able to get updates in case the happen.
Honestly I never tried SSR with glamorous, but it should work 😅